Zend/zend_vm_def.h

```diff
@@ -3470,6 +3470,13 @@ ZEND_VM_HOT_OBJ_HANDLER(112, ZEND_INIT_METHOD_CALL, CONST|TMPVAR|UNUSED|THIS|CV,
 		}
 	}
 
+	if (UNEXPECTED((fbc->common.fn_flags & ZEND_ACC_CTOR) != 0)) {
```

Comment:
This should be part of the `else` above, to reduce cost.

It might make sense to actually move this into `get_method`, which will prevent acquiring `__construct` as an ordinary method completely. But maybe that will interfere with `parent::__construct`?

Right now this is probably missing constructor calls via call_user_func.

Reply:
The reason why I left it outside of the else was that even though the CONST case should be caught by the compile time check, I wasn't completely sure if it would cover all the cases. I'll try to check those other cases/options.